powershell-yaml.psm1

```diff
@@ -290,17 +337,37 @@ function ConvertTo-Yaml {
             $wrt = New-Object "System.IO.StringWriter"
         }
 
+        $builder = New-Object "YamlDotNet.Serialization.SerializerBuilder"
+    
         if ($PSCmdlet.ParameterSetName -eq 'NoOptions') {
-            $Options = 0
-
             if ($JsonCompatible) {
                 # No indent options :~(
-                $options = [YamlDotNet.Serialization.SerializationOptions]::JsonCompatible
+                $builder = $builder.JsonCompatible()
+            }
+        } else {
+            # SerializationOptions is no longer used by YamlDotNet, translate it
+
+            if ($options.HasFlag([YamlDotNet.Serialization.SerializationOptions]::Roundtrip)) {
```

Comment:
You can try a switch here. Something like:

```powershell
switch ( $Options ) {
        EmitDefaults {
            $builder = $builder.EmitDefaults().Build()
        }
        DisableAliases {
            $builder = $builder.DisableAliases().Build()
        }
        JsonCompatible {
            $builder = $builder.JsonCompatible().Build()
        }
        Roundtrip {
            $builder = $builder.EnsureRoundtrip().Build()
        }
        DefaultToStaticType {
            $builder = $builder.WithTypeResolver((New-Object "YamlDotNet.Serialization.TypeResolvers.StaticTypeResolver"))
        }
        default {
            return $builder.Build()
        }
    }
```

But I think it would be cleaner to just create a separate function to take care of the serializer:

```powershell
function Get-Serializer {
    Param(
        [Parameter(Mandatory=$true)]
        [YamlDotNet.Serialization.SerializationOptions]$Options
    )
    $builder = New-Object "YamlDotNet.Serialization.SerializerBuilder"
    $builder = [StringQuotingEmitter]::Add($builder)
    switch ( $Options ) {
        EmitDefaults {
            return $builder.EmitDefaults().Build()
        }
        DisableAliases {
            return $builder.DisableAliases().Build()
        }
        JsonCompatible {
            return $builder.JsonCompatible().Build()
        }
        Roundtrip {
            return $builder.EnsureRoundtrip().Build()
        }
        DefaultToStaticType {
           return $builder.WithTypeResolver((New-Object "YamlDotNet.Serialization.TypeResolvers.StaticTypeResolver"))
        }
        default {
            return $builder.Build()
        }
    }
}
```

And I think this will also make the distinction between ```$null``` element and emty string.

Reply:
I'll create the `Get-Serializer` function, but `SerializationOptions` is a flags enumeration so more than one option could be set at the same time.  Using HasFlag handles the case where someone passes `JsonCompatible|Roundtrip`, for example.

Reply:
good point!